ygen/protogen.go

```diff
@@ -426,11 +566,17 @@ func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *gen
 					// child packages.
 					childpath := importPath(cfg.baseImportPath, cfg.basePackageName, childpkg)
 					if _, ok := imports[childpath]; !ok {
-						imports[childpath] = true
+						if !cfg.nestedMessages || msg.isFakeRoot {
+							imports[childpath] = true
+						}
 					}
 
 					p, _ := stripPackagePrefix(parentPkg, childpkg)
-					pfx = fmt.Sprintf("%s.", p)
+					if !cfg.nestedMessages || msg.isFakeRoot {
+						pfx = fmt.Sprintf("%s.", p)
+					} else {
+						pfx = ""
+					}
```

Comment:
remove else and put pfx = "" above if.

Reply:
Cleaned this all up - since we can do a negative check here and then just leave `pfx` unchanged rather than needing handling in this case.